File: 2366. Minimum Replacements to Sort the Array/cs/solution.cs
public class Solution {
    public long MinimumReplacement(int[] nums) {
        long steps = 0;

        for (int i = nums.Length-2; i >= 0; i--) {
            if (nums[i] <= nums[i+1]) continue;

            long stepsNeeded = (nums[i] + nums[i + 1] - 1) / nums[i + 1];
            steps += stepsNeeded - 1;
            nums[i] = (int)(nums[i] / stepsNeeded);
        }

        return steps;
    }
}

File: 2366. Minimum Replacements to Sort the Array/js/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumReplacement = function (nums) {
    let steps = 0;

    for (let i = nums.length - 2; i >= 0; i--) {
        if (nums[i] <= nums[i + 1]) continue;

        const stepsNeeded = Math.floor((nums[i] + nums[i + 1] - 1) / nums[i + 1]);
        steps += stepsNeeded - 1;
        nums[i] = Math.floor(nums[i] / stepsNeeded);
    }

    return steps;
};